### ISSUE TITLE
Unable to parse body with photo attached from iPhone

### ISSUE THREAD

**Issue by CryDeTaan** [php-mime-mail-parser/php-mime-mail-parser#362]
**Describe the bug**
When attaching a photo inline from iPhone the parser is unable to extract the body as it stops after the first boundary.

**To Reproduce**
I have provided two mime files to test with(`working.mime` and `not_working.mime`), as well as two php file(`test_working.php` and `test_not_working.php`) to test(based off instructions, so nothing complicated).
I had to add the .txt to all the files to attach to the Bug report, please remove the .txt extension to make them work :)

**Expected behaviour**
I am not sure if this would be possible, but I am guessing the `Content-Type: text/plain;` should be combined to make up the complete body text. 

**Screenshots**
![image](https://user-images.githubusercontent.com/11268952/110244753-2c2e1800-7f58-11eb-8765-d1f26c562f85.png)

**Version (please complete the following information):**
 - Version of mailparse: 3.1.1
 - Version of PHP PHP 7.4
 - Version of php-mime-mail-parser 7.0

[composer.json.txt](https://github.com/php-mime-mail-parser/php-mime-mail-parser/files/6097361/composer.json.txt)
[composer.lock.txt](https://github.com/php-mime-mail-parser/php-mime-mail-parser/files/6097362/composer.lock.txt)
[not_working.mime.txt](https://github.com/php-mime-mail-parser/php-mime-mail-parser/files/6097363/not_working.mime.txt)
[test_not_working.php.txt](https://github.com/php-mime-mail-parser/php-mime-mail-parser/files/6097364/test_not_working.php.txt)
[test_working.php.txt](https://github.com/php-mime-mail-parser/php-mime-mail-parser/files/6097365/test_working.php.txt)
[working.mime.txt](https://github.com/php-mime-mail-parser/php-mime-mail-parser/files/6097366/working.mime.txt)

**Comment by eXorus** [comment]
Thanks for your very clear report.

I already know this bug but it is not an easy one.

https://github.com/php-mime-mail-parser/php-mime-mail-parser/issues/164
https://github.com/php-mime-mail-parser/php-mime-mail-parser/issues/163
https://github.com/php-mime-mail-parser/php-mime-mail-parser/issues/160

I can keep the bug and try to work on it but not an easy task.
The goal will be to concatenate all the txt part or html part and not take only the first one.

**Comment by CryDeTaan** [comment]
Sorry for the repost then @eXorus :)

I can imagine it may not be that easy, I which I was at a point in my skill level to be of more help.

**Comment by develth** [comment]
Does this Bug still exists?
I tested it with current version & get in both versions 

![Bildschirmfoto 2024-12-12 um 18 14 03](https://github.com/user-attachments/assets/7ff48e21-f934-460d-918e-dd4bdec22bd8)

**Comment by eXorus** [comment]
Yes, the bug still exists. There's a workaround: you can manually merge the two HTML or text parts yourself, but the library doesn't handle it automatically.
That's the improvement I'd like to implement.

Currently, if you have an HTML part, followed by an inline image, then a second HTML part:
In the current version, you only get the first HTML part and the inline image.
In the expected version, you should get a single HTML output that includes both HTML parts and the inline image.

**Comment by VentoZwerg** [comment]
I have an even more strange, probably related observation:

If, on an iPhone, I take a screenshot and select "Share -> E-Mail", the resultine E-Mail shows completely empty with no attachment to the library. Some Apple E-Mails seem to have that "extra dimension" where sometimes there are attachments within attachments, and the library does not seem to cope well with it.

Any idea how to get around this?

PS: I can post .eml examples if anybody is interested...
